src/runtime/methodbinder.cs

```diff
@@ -275,19 +282,40 @@ internal static MethodInfo MatchSignatureAndParameters(MethodInfo[] mi, Type[] g
 
                     for (int n = 0; n < clrnargs; n++) {
                         IntPtr op;
-                        if (n < pynargs)
-                        {
-                            if (arrayStart == n)
-                            {
+                        if (n < pynargs) {
+                            if (arrayStart == n) {
                                 // map remaining Python arguments to a tuple since
                                 // the managed function accepts it - hopefully :]
                                 op = Runtime.PyTuple_GetSlice(args, arrayStart, pynargs);
                             }
-                            else
-                            {
+                            else {
                                 op = Runtime.PyTuple_GetItem(args, n);
                             }
-                            Type type = pi[n].ParameterType;
+                            type = null;
+                            if  (_methods.Length>1) { 
+                                IntPtr pyoptype = IntPtr.Zero;
+                                try {
+                                    pyoptype = Runtime.PyObject_Type(op);
+                                }
+                                catch (System.Exception e) {
```

Comment:
Does PyObject_Type ever throw an Exception? Doesn't it just return NULL on failure?